modules/lib/src/main/resources/assets/js/app/inputtype/ui/text/dialog/LinkModalDialog.ts

```diff
@@ -888,10 +897,13 @@ export class LinkModalDialog
 
             const fragment = anchorString ? `${LinkModalDialog.fragmentPrefix}${anchorString}`: '';
 
+            if (!fragment) {
+                return StringHelper.EMPTY_STRING;
+            }
+
             return getQueryParamsString() ? '&' + fragment : '?' + fragment;
         };
 
-
         const setUrl = () => {
```

Comment:
extract all internal functions to private class methods

setUrl name is confusing since you're not only setting url within it